<a href="https://colab.research.google.com/github/Bima34157/data-science-2026/blob/main/pertemuan_2_zaldi_bima_aditya_220401010271.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Pertemuan ke-2

- Nama    : Zaldi Bima Aditya
- NIM     : 2220401010271
- Kelas   : IF405

In [2]:
import pandas as pd
import numpy as np

url = "https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv"
df = pd.read_csv(url)

In [4]:
df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


# **Jawaban Praktikum Pertemuan 2**

**Q1. Berapa jumlah baris dan kolom pada dataset Titanic? Kolom apa saja yang tersedia?**

In [6]:
print(f"Shape  : {df.shape}")
print(f"Baris  : {df.shape[0]}")
print(f"Kolom  : {df.shape[1]}")
print("\nDaftar kolom:")
for i, col in enumerate(df.columns, 1):
    print(f"  {i:2}. {col}")

Shape  : (891, 12)
Baris  : 891
Kolom  : 12

Daftar kolom:
   1. PassengerId
   2. Survived
   3. Pclass
   4. Name
   5. Sex
   6. Age
   7. SibSp
   8. Parch
   9. Ticket
  10. Fare
  11. Cabin
  12. Embarked


Jawaban:

Dataset Titanic memiliki 891 baris dan 12 kolom.

**Q2. Kolom mana saja yang memiliki missing values? Berapa jumlah dan proporsinya (%) per kolom?**

In [7]:
missing = df.isnull().sum()
percent = (missing / len(df)) * 100

pd.DataFrame({
    'Missing': missing,
    'Percent (%)': percent
})

,Missing,Percent (%)
PassengerId,0,0.000000
Survived,0,0.000000
Pclass,0,0.000000
Name,0,0.000000
Sex,0,0.000000
Age,177,19.865320
SibSp,0,0.000000
Parch,0,0.000000
Ticket,0,0.000000
Fare,0,0.000000


Jawaban:

Berikut adalah kolom yang memiliki missing value:

1. Kolom Cabin, sebanyak 687 baris kosong (77.1%)
2. Kolom Age,sebanyak 177 baris kosong (19.9%)
3. Kolom Embarked, sebanyak 2 baris kosong (0.2%)


---



**Q3. Berapa persen penumpang yang selamat (Survived=1)? Gunakan operasi NumPy atau Pandas.**

In [12]:

# Metode 1: Menggunakan Pandas
persen_selamat_pd = (df['Survived'].sum() / len(df)) * 100
jumlah_selamat_pd = df['Survived'].sum()
jumlah_tidak_selamat = len(df) - jumlah_selamat_pd

# Metode 2: Menggunakan NumPy
survived_array = np.array(df['Survived'])
persen_selamat_np = (np.sum(survived_array) / len(survived_array)) * 100

print(f"\nJumlah penumpang selamat: {int(jumlah_selamat_pd)}")
print(f"Jumlah penumpang tidak selamat: {int(jumlah_tidak_selamat)}")
print(f"Persentase selamat (Pandas): {persen_selamat_pd:.2f}%")
print(f"Persentase selamat (NumPy): {persen_selamat_np:.2f}%")

# Visualisasi tabel
survival_count = df['Survived'].value_counts()
print(f"\nRingkasan:")
print(f"  Tidak selamat: {survival_count.get(0, 0)} penumpang")
print(f"  Selamat: {survival_count.get(1, 0)} penumpang")


Jumlah penumpang selamat: 342
Jumlah penumpang tidak selamat: 549
Persentase selamat (Pandas): 38.38%
Persentase selamat (NumPy): 38.38%

Ringkasan:
  Tidak selamat: 549 penumpang
  Selamat: 342 penumpang


**Q4. Filter penumpang wanita dari kelas 1 (Pclass=1). Berapa jumlahnya dan berapa rata-rata usianya?**

In [13]:
# ============================================================
# Q4. Filter penumpang wanita dari kelas 1 (Pclass=1). Berapa jumlahnya dan berapa rata-rata usianya?
# ============================================================
wanita_kelas1 = df[(df['Sex'] == 'female') & (df['Pclass'] == 1)]
jumlah = len(wanita_kelas1)
rata_usia = wanita_kelas1['Age'].mean()

print(f"\nJumlah penumpang wanita kelas 1: {jumlah}")
print(f"Rata-rata usia: {rata_usia:.2f} tahun")
print(f"\nDetail penumpang wanita kelas 1:")
print(wanita_kelas1[['Name', 'Age', 'Sex', 'Pclass', 'Survived']].to_string(index=False))


Jumlah penumpang wanita kelas 1: 94
Rata-rata usia: 34.61 tahun

Detail penumpang wanita kelas 1:
                                                                              Name  Age    Sex  Pclass  Survived
                               Cumings, Mrs. John Bradley (Florence Briggs Thayer) 38.0 female       1         1
                                      Futrelle, Mrs. Jacques Heath (Lily May Peel) 35.0 female       1         1
                                                          Bonnell, Miss. Elizabeth 58.0 female       1         1
                                    Spencer, Mrs. William Augustus (Marie Eugenie)  NaN female       1         1
                                          Harper, Mrs. Henry Sleeper (Myna Haxtun) 49.0 female       1         1
                                                               Icard, Miss. Amelie 38.0 female       1         1
                                                        Fortune, Miss. Mabel Helen 23.0 female       1        

Dari dataset diatas, penumpang wanita dari kelas 1 sebanyak 94 penumpang, dengan rata-rata usia 34.61 tahun.

**Q5 Gunakan groupby untuk menghitung tingkat keselamatan (mean Survived) per kelas penumpang. Kelas mana yang paling tinggi?**

In [14]:
hasil1= df.groupby('Pclass')['Survived'].mean()
print(hasil1)

Pclass
1    0.629630
2    0.472826
3    0.242363
Name: Survived, dtype: float64


**Q6 Buat analisis tambahan berdasarkan variabel lain yang Anda anggap menarik. Sertakan interpretasi singkat.**

In [15]:
# ============================================================
# Analisis Tambahan: Gender dan Kelas Penumpang
# ============================================================

pivot = (
    df.groupby(['Pclass', 'Sex'])['Survived']
    .mean()
    .unstack()
    .round(3) * 100
)

print("Persentase Keselamatan berdasarkan Kelas dan Gender (%)")
print(pivot)

print("\n--- Interpretasi ---")
print("""
Hasil menunjukkan bahwa:

- Wanita memiliki tingkat keselamatan lebih tinggi pada semua kelas.
- Wanita kelas 1 memiliki peluang keselamatan tertinggi.
- Pria kelas 3 memiliki peluang keselamatan terendah.
- Selain gender, status sosial yang direpresentasikan oleh kelas
  penumpang juga berpengaruh terhadap peluang keselamatan.
""")

Persentase Keselamatan berdasarkan Kelas dan Gender (%)
Sex     female  male
Pclass              
1         96.8  36.9
2         92.1  15.7
3         50.0  13.5

--- Interpretasi ---

Hasil menunjukkan bahwa:

- Wanita memiliki tingkat keselamatan lebih tinggi pada semua kelas.
- Wanita kelas 1 memiliki peluang keselamatan tertinggi.
- Pria kelas 3 memiliki peluang keselamatan terendah.
- Selain gender, status sosial yang direpresentasikan oleh kelas
  penumpang juga berpengaruh terhadap peluang keselamatan.

